File: OP_lab/lab5/c#/Square.cs
using System;

namespace lab_5_OP
{
    public class Square : Figure
    {
        public Square(Tuple<double, double> c1, Tuple<double, double> c2)
        {
            _tops[0] = c1;
            _tops[1] = c2;
        }
        
        public override double Perimeter()
        {
            return 4.0 * Math.Sqrt(this.Area());
        }

        public override double Area()
        {
            return (Math.Pow(_tops[0].Item1 - _tops[1].Item1, 2) +
                    Math.Pow(_tops[0].Item2 - _tops[1].Item2, 2)) / 2.0;
        }

        private readonly Tuple<double, double>[] _tops = new Tuple<double, double>[2];
    }
}

File: OP_lab/lab5/c++/square.cpp
//
// Created by maxim on 25.03.2020.
//

#include "square.h"

square::square(Coord c1, Coord c2) : tops{c1, c2}{
}

double square::perimeter() {
    return 4.0 * sqrt(this->area());
}

double square::area() {
    return (pow(tops[0].first - tops[1].first, 2) +
            pow(tops[0].second - tops[1].second, 2)) / 2.0;
}
